customerio-reactnative-richpush.podspec

```diff
@@ -0,0 +1,32 @@
+require "json"
+
+package = JSON.parse(File.read(File.join(__dir__, "package.json")))
+
+# Used by customers to install native iOS dependencies inside their Notification Service Extension (NSE) target to setup rich push. 
+# Note: We need a unique podspec for rich push because the other podspecs in this project install too many dependencies that should not be installed inside of a NSE target. We need this podspec which installs minimal dependencies that are only included in the NSE target. 
+Pod::Spec.new do |s|
+  s.name         = "customerio-reactnative-richpush"
+  s.version      = package["version"]
+  s.summary      = package["description"]
+  s.homepage     = package["homepage"]
+  s.license      = package["license"]
+  s.authors      = package["author"]
+
+  s.platforms    = { :ios => "13.0" }
+  s.source       = { :git => "https://github.com/customerio/customerio-ios.git", :tag => "#{s.version}" }
+
+  s.source_files = "ios/**/*.{h,m,mm,swift}"
+
+  # s.dependency "X", "X"
+  # Careful when declaring dependencies here. All dependencies will be included in the App Extension target in Xcode, not the host iOS app.   
+
+  # Subspecs allow customers to choose between multiple options of what type of version of this rich push package they would like to install. 
+  s.subspec 'apn' do |ss|
+    ss.dependency "CustomerIO/MessagingPushAPN", package["cioNativeiOSSdkVersion"]
+  end
+
+  s.subspec 'fcm' do |ss|
+    ss.dependency "CustomerIO/MessagingPushFCM", package["cioNativeiOSSdkVersion"]
+    ss.dependency "FirebaseMessaging"
```

Comment:
Should we add this dependency? we might encounter the same issue we did for iOS where we couldn't keep up with FCM versions. 

Maybe, let the user add it themselves?

Reply:
I have had some of the same concerns myself. My biggest reason for adding it is that our RN docs today tell you to use this exact syntax in your `Podfile`. Because of that, I don't see much difference between a customer adding it and us adding it ourselves. 

This brings up something that is potentially more important. Just because we allow customers to control what *version* of FCM is installed doesn't mean that our native SDK is compatible with that version. 

I think we should consider including the FCM SDK as a dependency of `CustomerIO/MessagingPushFCM`. That would not only make setup easier for RN customers but also native iOS customers. 

Customers will still have the ability, if they choose, to use a specific version of the FCM SDK if they want to. We could make our FCM version requirement very flexible to allow that. 

It will not take me long, so let me make a draft PR for iOS and we can discuss the solution there.

Reply:
[Conversation in Slack](https://customerio.slack.com/archives/C01QYDKD0SU/p1686852783619059) talking about this change.

Reply:
I added some thoughts on the change on the Slack thread, they would still apply here as well.

Reply:
I prefer to hold this PR until the discussion is finished. It's a good discussion that impacts customers and are all easy changes to make no matter what we decide.

Reply:
iOS SDK 2.7.0 includes FCM as a dependency now. Therefore, we are no longer blocked and can continue with this PR.